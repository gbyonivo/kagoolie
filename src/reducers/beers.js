import { FETCH_BEERS, FINISHED_FETCHING_BEERS, ERROR_FETCHING_BEERS } from '../constants/actionTypes';

const initialState = {
  beers: [],
  errors: [],
  isFetchingBeers: true
};

const ACTION_HANDLERS = {
  [FETCH_BEERS]: state => ({
    ...state,
    isFetchingBeers: true
  }),
  [FINISHED_FETCHING_BEERS]: (state, { beers }) => ({
    ...state,
    beers,
    isFetchingBeers: false
  }),
  [ERROR_FETCHING_BEERS]: (state, { error }) => ({
    ...state,
    errors: [...state.errors, error],
    isFetchingBeers: false
  })
};

export default (state = initialState, { payload, type }) => {
  const handler = ACTION_HANDLERS[type];
  return handler ? handler(state, payload) : state;
};