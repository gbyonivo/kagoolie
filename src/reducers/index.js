import { combineReducers } from 'redux';
import beersReducer from './beers';

export default () => combineReducers({
  beersReducer
});