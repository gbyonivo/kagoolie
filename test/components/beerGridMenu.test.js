import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducers from '../__testData__/store';
import combinedReducer from '../../src/reducers';
import BeerGridMenu from '../../src/components/beerGridMenu';

describe('Beer Grid Menu component', () => {
  it('renders correctly', () => {
    const store = createStore(combinedReducer(), reducers);
    const actual = renderer.create(<Provider store={store}>
      <BeerGridMenu />
    </Provider>);
    expect(actual).toMatchSnapshot();
  });
});