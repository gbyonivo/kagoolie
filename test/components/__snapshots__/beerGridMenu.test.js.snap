// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beer Grid Menu component renders correctly 1`] = `
<div
  className="beerGridMenu"
>
  <div
    className="selectInputWrapper"
  >
    <label
      className="selectInputLabel"
    >
      Alcohol Volume
    </label>
    <select
      className="selectInput"
      name="gridFilter"
      onChange={[Function]}
      value="LOW_TO_HIGH"
    >
      <option
        value="LOW_TO_HIGH"
      >
        LOW_TO_HIGH
      </option>
      <option
        value="HIGH_TO_LOW"
      >
        HIGH_TO_LOW
      </option>
    </select>
  </div>
  <div
    className="beerGridMenuItems"
  >
    <div
      className="beerGridMenuItem"
    >
      <div
        className="image"
      >
        <img
          alt=""
          src="sdsds"
        />
      </div>
      <div
        className="name"
      >
        jupiter
      </div>
    </div>
    <div
      className="beerGridMenuItem"
    >
      <div
        className="image"
      >
        <img
          alt=""
          src="sdsds"
        />
      </div>
      <div
        className="name"
      >
        pluto
      </div>
    </div>
  </div>
</div>
`;
